test/hooks.ts

```diff
@@ -270,19 +271,19 @@ test('`afterResponse` hook is called with request, normalized options, and respo
 				json,
 				hooks: {
 					afterResponse: [
-						// @ts-expect-error
 						async (request, options, response) => {
 							if (response.status === 403) {
 								// Retry request with valid token
 								return ky(request, {
 									...options,
 									json: {
-										// @ts-expect-error @TODO
-										...options.json,
+										...(options as Options).json,
```

Comment:
we cannot use spread here if it's not an object

if it's valid only in test, let's cast it here

Reply:
The real issue is that the `options` parameter doesn't have the correct type.

Reply:
What is the expected type here?

Reply:
Also, should it be connected to the `json`  we have in test?

In this case we need to have Generic type:

It will be `Options<Json extends BasicJson = BasicJson>` instead of `Options`

What do you think?

Reply:
@sindresorhus I can suggest these changes regarding more stricter mapping of `Json` and the types in hooks and `json`

You can have a look at the changes (I didn't finish it, but you can already see the possible way) – https://github.com/Beraliv/ky/pull/1

wdyt?

Reply:
I think `...options.json as Record<string, unknown>` would be fine here. We know it's an object, so it's fine to force cast.

Reply:
Do you think https://github.com/Beraliv/ky/pull/1 will be too much?

Reply:
I think:

> Instead of ky.get(...).json<IncomingHttpHeaders>() we will have ky.get<IncomingHttpHeaders>(...).json()

is a good idea, but making it extend a JSON type is problematic for many reasons. TS is just not good enough to make it workable. https://github.com/sindresorhus/ky/pull/133#discussion_r282433150

Reply:
Okay, let's try it in the future if interested